code/lib/components/src/icon/icon.tsx

```diff
@@ -1,39 +1,47 @@
-import React, { ComponentProps, memo } from 'react';
+import React, { FunctionComponent, ComponentProps, memo } from 'react';
 import { styled } from '@storybook/theming';
-import icons, { IconKey } from './icons';
+import { icons, IconKey } from './icons';
 
-import Svg from './svg';
+const Svg = styled.svg`
+  display: inline-block;
+  shape-rendering: inherit;
+  transform: translate3d(0, 0, 0);
+  vertical-align: middle;
 
-const Path = styled.path({
-  fill: 'currentColor',
-});
+  path {
+    fill: currentColor;
+  }
+`;
 
 export interface IconsProps extends ComponentProps<typeof Svg> {
-  icon?: IconKey;
-  symbol?: IconKey;
+  icon: IconType;
+  useSymbol?: boolean;
```

Comment:
This seems like a breaking change to me. People do use this component when building their own addons, right? If that is the case, we need to document this in the MIGRATION document.

@shilman can you confirm?

Reply:
Yes, we should document this in MIGRATION.md! 💯

Reply:
@yannbf @shilman Can you of you help me document this?

Reply:
Done ✅